draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,551 @@ author:
-sf-integer. Alias forms, with priority 0, MUST NOT be included.
+sf-integer.
 
 The TTL of the record MUST be a parameter with the key "ttl", and a value as an sf-integer.
+This value must be the minimum TTL value of any CNAME or SVCB record encountered while
+resolving this SVCB record.
+
+SvcParams are represented with a parameter string constructed
+prepending the string "key" to the numeric version of the
+SvcParamKey. For example, the ALPN SvcParamKey, with the numeric value
+1, would have a parameter key "key1". The value of each parameter MUST
+be an sf-binary item that contains the bytes of the SvcParamValue.
+
+If Proxy-DNS-Request included a "params" parameter, the proxy server
+MAY filter the SvcParams to only include SvcParams whose keys were
+included in the "params" list.  Severs MAY include additional
+SvcParams.  In particular, servers SHOULD include the "mandatory"
+parameter if present, which would be presented as "key0", along with
+any parameters that are defined as mandatory for that record.
+
+Proxy servers MUST NOT include the "Proxy-DNS-SVCB" response header field if the
+corresponding request did not include a "Proxy-DNS-Request".
+
+Proxy servers MUST attempt to resolve the SVCB DNS query name to
+obtain the SVCB RRset to return in this header.  If this resolves to a
+SVCB AliasMode record, proxy servers MUST resolve the TargetName of
+that AliasMode record to obtain a ServiceMode record.  Both
+resolutions MUST use the same RR type specified in the "t" parameter
+of Proxy-DNS-Request.
+Which records are returned in this header is determined by:
+
+* If the SVCB DNS query name resolves to a SVCB ServiceMode record,
+  only the ServiceMode records SHALL be included in the Proxy-DNS-SVCB
+  list.
+
+* If the SVCB DNS query name resolves to a SVCB AliasMode record, and
+  if the TargetName of the SVCB AliasMode record resolves to a SVCB
+  ServiceMode RRSet, only the ServiceMode records SHALL be included in
+  the Proxy-DNS-SVCB list.
+  
+* If the SVCB DNS query name resolves to a SVCB AliasMode record, and
+  if the TargetName of a SVCB AliasMode record does not resolve to any
+  ServiceMode records, the Proxy-DNS-SVCB list SHALL include a single
+  item containing the AliasMode TargetName and "priority=0".
+
+* If no SVCB records are found authoritatively (i.e., a "NOERROR" or
+  "NXDOMAIN" response when resolving the SVCB DNS query name specified
+  in the Proxy-DNS-Request header), the list MUST contain a single
+  entry with the item value of ".", indicating that the name has no
+  SVCB records.  This special value of "." MUST NOT be returned in
+  error conditions such as timeouts.  The value of the "ttl" parameter
+  MUST be the TTL of this authoritative DNS response.
+
+* If an error condition is encountered (such as a timeout, loop, or
+  invalid record), the Proxy-SVCB-DNS header MUST contain an empty list
+  (and thus not be returned).
```

Comment:
I'm not sure these are the same thing.  How about simply....
```suggestion
  invalid record), the Proxy-SVCB-DNS header MUST NOT be returned.
```

Reply:
In structured headers they are the same thing.  Would saying an empty sf-list be more clear?